### PR TITLE
Fix/qdrant tests in CI

### PR DESCRIPTION
**Pull request by steinitzu**
### Description

* Fixes issues with qdrant state restore
* Support local qdrant
* Test both local and server qdrant (same tests as weaviate)
*  server qdrant runs in docker in CI instead of cloud

### Related Issues

- Resolves https://github.com/dlt-hub/dlt/issues/1337

### CODE REVIEW

File: dlt/destinations/impl/qdrant/factory.py [hunk 1]
```diff
@@ -26,6 +28,20 @@ def _raw_capabilities(self) -> DestinationCapabilitiesContext:
...
+        cls,
+        caps: DestinationCapabilitiesContext,
+        config: QdrantClientConfiguration,
+        naming: t.Optional[NamingConvention],
+    ) -> DestinationCapabilitiesContext:
+        caps = super(qdrant, cls).adjust_capabilities(caps, config, naming)
+        if config.credentials.is_local():
+            # Local qdrant can not load in parallel
+            caps.loader_parallelism_strategy = "sequential"
+            caps.max_parallel_load_jobs = 1
```
Comment (steinitzu) [inline dlt/destinations/impl/qdrant/factory.py:42]: workaround for local qdrant, it is not thread safe

File: dlt/load/load.py [hunk 2]
```diff
@@ -196,7 +196,9 @@ def w_spool_job(
     def spool_new_jobs(self, load_id: str, schema: Schema) -> Tuple[int, List[LoadJob]]:
         # use thread based pool as jobs processing is mostly I/O and we do not want to pickle jobs
         load_files = filter_new_jobs(
-            self.load_storage.list_new_jobs(load_id), self.destination.capabilities(), self.config
+            self.load_storage.list_new_jobs(load_id),
+            self.destination.capabilities(self.initial_client_config),
```
Comment (steinitzu) [inline dlt/load/load.py:200]: I had to pass config here to get the correct adjusted capabilities (see `qdrant/factory.py` changes). Please double check.

File: tests/load/pipeline/test_restore_state.py [hunk 3]
```diff
@@ -65,9 +65,10 @@ def test_restore_state_utils(destination_config: DestinationTestConfiguration) -
     with p.destination_client(p.default_schema.name) as job_client:  # type: ignore[assignment]
         with pytest.raises(DestinationUndefinedEntity):
             load_pipeline_state_from_destination(p.pipeline_name, job_client)
-        # sync the schema
-        p.sync_schema()
-        # check if schema exists
+    # sync the schema
+    p.sync_schema()
+    # check if schema exists
```
Comment (steinitzu) [inline tests/load/pipeline/test_restore_state.py:70]: Local qdrant uses flock and can't handle more than one instance of client, why I moved all the pipeline operations out of `with destination_client()` scope

File: dlt/destinations/impl/qdrant/factory.py (hunk 1 above)
Comment (rudolfix) [reply dlt/destinations/impl/qdrant/factory.py:42]: very true! this is why this capability got added :)

File: dlt/common/destination/reference.py [hunk 4]
```diff
@@ -63,6 +63,28 @@ class StorageSchemaInfo(NamedTuple):
     inserted_at: datetime.datetime
     schema: str
 
+    @classmethod
+    def from_normalized_mapping(
```
Comment (rudolfix) [inline dlt/common/destination/reference.py:67]: this is good!

**Review by rudolfix — DISMISSED** [review]
LGTM! I can't find where you test local (not server) quadrant. the test suite seems to be run only once

File: dlt/load/load.py (hunk 2 above)
Comment (rudolfix) [reply dlt/load/load.py]: I even improved it and requested full (not just initial) config here

**Comment by steinitzu** [comment]
> LGTM! I can't find where you test local (not server) quadrant. the test suite seems to be run only once

I just added two destination configs so they both run together. (see `local-file`/`server` suffix)

```
tests/load/pipeline/test_arrow_loading.py::test_load_arrow_with_not_null_columns[arrow-batch-qdrant-no-staging-local-file] PASSED [ 33%]
tests/load/pipeline/test_arrow_loading.py::test_load_arrow_with_not_null_columns[arrow-batch-qdrant-no-staging-server] PASSED [ 33%]
```

https://github.com/dlt-hub/dlt/pull/1526/files#diff-136cf9e252c2cfe3f09641909b446ac65b5d00eb9286caf484bc5f1032437988R288-R293

**Pull request merged**